frontend/src/widgets/resultaggregator.js

```diff
@@ -22,176 +22,166 @@ import WidgetHeader from '../components/widget-header';
-    ];
-    return (
-      <Card>
-        <WidgetHeader title={this.title} getDataFunc={this.getResultData} onEditClick={this.props.onEditClick} onDeleteClick={this.props.onDeleteClick}/>
-        <CardBody data-id="recent-result-data">
-          {this.state.resultAggregatorError &&
-            <p>Error fetching data</p>
-          }
-          {(this.state.total === 0 && !this.state.isLoading) &&
-            <p>No data returned, try changing the days.</p>
-          }
-          {(!this.state.resultAggregatorError && this.state.isLoading) &&
-          <Text component="h2">Loading ...</Text>
-          }
-          {(!this.state.resultAggregatorError && !this.state.isLoading && this.params.chart_type === 'pie' && this.state.total !== 0) &&
-            <ChartPie
-              constrainToVisibleArea={true}
-              data={this.state.chartData}
-              legendData={this.state.legendData}
-              labels={({datum}) => `${toTitleCase(datum.x)}: ${datum.y}`}
-              labelComponent={
-                <ChartTooltip
-                  constrainToVisibleArea
-                  dx={-10}
-                  style={{ fill: 'white' }}
-                />
-              }
-              width={350}
-              padding={{
-                bottom: 20,
-                left: 20,
-                right: 20,
-                top: 20
-              }}
-              themeColor={ChartThemeColor.multi}
-            />
-          }
-          {(!this.state.resultAggregatorError && !this.state.isLoading && this.params.chart_type === 'donut' && this.state.total !== 0) &&
-            <ChartDonut
-              constrainToVisibleArea
-              data={this.state.chartData}
-              legendData={this.state.legendData}
-              labels={({datum}) => `${toTitleCase(datum.x || '')}: ${datum.y}`}
-              height={200}
-              title={this.state.total}
-              subTitle="total results"
-              style={{
-                labels: {fontFamily: 'RedHatText'}
-              }}
-              colorScale={themeColors}
-            />
-          }
-        </CardBody>
-        <CardFooter>
-          {!this.state.isLoading && this.state.total !== 0 &&
-          <ChartLegend
-            data={this.state.legendData}
-            height={120}
-            orientation="horizontal"
-            responsive={false}
-            itemsPerRow={2}
-            colorScale={themeColors}
-            style={{
-              labels: {fontFamily: 'RedHatText'},
-              title: {fontFamily: 'RedHatText'}
+  const themeColors = [
+    'var(--pf-v5-global--success-color--100)',
+    'var(--pf-v5-global--danger-color--100)',
+    'var(--pf-v5-global--warning-color--100)',
+    'var(--pf-v5-global--info-color--100)',
+    'var(--pf-v5-global--primary-color--100)'
+  ];
+
+  return (
+    <Card>
+      <WidgetHeader title={title} getDataFunc={getChartData} onEditClick={onEditClick} onDeleteClick={onDeleteClick}/>
```

Comment:
If you move `getChartData` to a direct `useEffect` hook, this prop would be removed and `WidgetHeader` just won't render a refresh button.

Reply:
Done